### PR TITLE
New IMA server

### PR DESCRIPTION
**Pull request by mjancarik**

### CODE REVIEW

File: packages/server/lib/factory/IMAInternalFactory.js [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+    return router.route(router.getPath());
+  }
+
+  return {
+    _initApp,
+    createBootConfig,
+    _importAppMainSync,
+    _createDummyApp,
+    _getRouteInfo,
+    _addImaToResponse,
```
Comment (jsimck) [inline packages/server/lib/factory/IMAInternalFactory.js:255]: why do these have `_` to discourage user from using it outside of the server pkg?

File: packages/server/lib/factory/IMAInternalFactory.js [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+    res.$IMA = res.$IMA ? { ...res.$IMA, routeName } : { routeName };
+  }
+
+  function _importAppMainSync({ environment, context = {} }) {
+    let appMain = serverGlobal.has(GLOBAL.APP_MAIN)
+      ? serverGlobal.get(GLOBAL.APP_MAIN)
+      : appFactory();
+
+    if (environment.$Env === 'dev') {
+      appMain = appFactory();
```
Comment (jsimck) [inline packages/server/lib/factory/IMAInternalFactory.js:151]: doesn't this call `appFactory()` twice if it's not already in the `serverGlobals`?

File: packages/server/lib/factory/IMAInternalFactory.js [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+      console.warn('Failed to retrieve current route info', {
+        error: e,
+      });
+    }
+
+    return routeInfo;
+  }
+
+  function _addImaToResponse({ req, res }) {
+    let routeName = 'other';
```
Comment (jsimck) [inline packages/server/lib/factory/IMAInternalFactory.js:135]: I know it was already there... but what's the point of `'other'` ?

File: packages/server/lib/factory/IMAInternalFactory.js (hunk 1 above)
Comment (mjancarik) [reply packages/server/lib/factory/IMAInternalFactory.js:255]: only for defined interal functions

File: packages/server/lib/factory/IMAInternalFactory.js (hunk 3 above)
Comment (mjancarik) [reply packages/server/lib/factory/IMAInternalFactory.js:135]: only for keeping routeName as string..

File: package.json [hunk 4]
```diff
@@ -29,7 +29,7 @@
     "doc": "npm run build:doc --workspace website",
     "changeset": "changeset add",
     "postinstall": "npm run build",
-    "lint": "eslint --cache --cache-location=node_modules/.cache/ './**/*.{js,jsx,ts,tsx,mjs}'",
+    "lint": "eslint --fix --cache --cache-location=node_modules/.cache/ './**/*.{js,jsx,ts,tsx,mjs}'",
```
Comment (Filipoliko) [inline package.json:32]: this command is also called as part of travis CI, the fix has to be either removed, or moved to a separated command

**Pull request merged**